notebooks/working/Simple_Simulation.md

```diff
@@ -0,0 +1,382 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,py,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.2'
+      jupytext_version: 1.3.4
+  kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+---
+
+```python
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from causalgraphicalmodels import CausalGraphicalModel, StructuralCausalModel
+import pylogit
+from collections import OrderedDict
+import pylogit as cm
+from functools import reduce
+import statsmodels.api as sm
+
+from IPython import display
+
+import os
+os.listdir('.')
+```
+
+```python
+data = pd.read_csv('spring_2016_all_bay_area_long_format_plus_cross_bay_col.csv')
+data.columns
+```
+
+```python
+### Just look at the drive alone mode
+drive_alone_df = data[data['mode_id']==1]
+
+### Only keep columns of interest 
+drive_alone_df = drive_alone_df[[ 'total_travel_time', 'total_travel_cost',
+                                 'total_travel_distance', 'household_size',
+                               'num_cars', 'cross_bay']]
+
+drive_alone_df.describe()
+
+```
+
+## Assumed causal graph
+
+```python
+drive_alone_graph = CausalGraphicalModel(
+    nodes=['total_travel_time', 'total_travel_cost', 'total_travel_distance', #'household_income',
+          'household_size',  'num_cars', 'cross_bay', 'utility_driving'],
+    edges=[
+        ("total_travel_time", "utility_driving"), 
+        ("total_travel_cost", "utility_driving"), 
+        ("total_travel_distance", "utility_driving"), 
+        ("household_size", "utility_driving"), 
+#         ("household_income", "utility_driving"), 
+        ("num_cars", "utility_driving"), 
+        ("cross_bay", "utility_driving"), 
+        
+        
+        ("total_travel_distance", "total_travel_time"), 
+        ("total_travel_distance", "total_travel_cost"), 
+    ]
+)
+
+# draw return a graphviz `dot` object, which jupyter can render
+drive_alone_graph.draw()
+```
+
+## Distributional regression
+Assume univariate linear approximation for the relationship between travel distance and travel time/cost. Turns out it's not a horrible assumption.
+
+```python
+def fit_regression(X, y, data, plotting = True):
+    data_x = sm.add_constant(data[X])
+    data_y = data[y]
+
+    model = sm.OLS(data_y, data_x)
+    
+    results = model.fit()
+    
+    if plotting:
+        fig = plt.figure(figsize=(12,8))
+        fig = sm.graphics.plot_regress_exog(results, X[0], fig=fig)
+    
+    return results
+```
+
+```python
+## Regress travel cost on travel distance
+
+cost_on_distance_reg = fit_regression(X=['total_travel_distance'], 
+                                      y = ['total_travel_cost'], data=drive_alone_df, plotting = True)
+
+cost_on_distance_reg.summary()
```

Comment:
Hmm, from the plots travel cost looks categorical (hence the distinct diagonal lines in the plot of the residuals vs travel distance. Perhaps you should use an MNL instead of a linear regression for predicting travel distance?